pkgs/os-specific/linux/system76-io/default.nix

```diff
@@ -1,6 +1,6 @@
 { lib, stdenv, fetchFromGitHub, kernel }:
 let
-  version = "1.0.2";
+  version = "1.0.4";
   sha256 = "sha256-DWUjQmoojkzFv1p4Xyt0kOwwqQ216ocO5yR/ujhhMPA=";
```

Comment:
When the version changes, the hash needs to be updated as well.  The easiest way to determine what the hash value should be is to set it to the empty string, then just attempt the build:

    cd nixpkgs.git
    nix-build . -QA linuxPackages.system76-io --no-out-link

The build will fail and give you the correct value to put in here.

This is an exceedingly common Nix footgun.  For most packages, if you bump the version but don't update the hash, the build will *silently* continue using the old source tarball.  This is because the package's derivation depends on a source derivation (i.e. containing the source tree) called e.g. `/nix/store/a36b9a0...-source` that is a *fixed-output derivation*, meaning that if that output hash + name already exists in the Nix store (and it will if you've already built the old version of the package at some point), then it just continues using it, despite the upstream source URL (i.e. the source derivation's inputs) changing (which would normally trigger Nix to rebuild something).  Fixed-output derivations are used for validation and so that URLs can be changed without triggering mass rebuilds.

Reply:
I had been using `nix-build -A packageName` for my COSMIC packages but lately using `nix-update` to make it less manual. I'll update this now.

Reply:
Sooo the build worked:

```
aaronh@drack:~/Projects/NixOS/nixpkgs$ nix-build . -QA linuxPackages.system76-io --no-out-link
this derivation will be built:
  /nix/store/iy5pvr57bir5dkgiyhfg529w61f6525s-system76-io-module-1.0.4-6.6.56.drv
these 11 paths will be fetched (255.03 MiB download, 871.71 MiB unpacked):
  /nix/store/wxgwfw64c1rmsi8nhdb0dz8s4haycj6b-elfutils-0.191-bin
  /nix/store/5zc6x2aylvgzi32qna61fshgi6abgvmx-elfutils-0.191-dev
  /nix/store/qswh3390mf0shsrbq54md3l8w3jxxcfd-getopt-1.1.6
  /nix/store/bbcq7nbvp2wqdjm1j2qxqlmpm3maqnx5-libffi-3.4.6
  /nix/store/9dvhqgc1bqsq7xf8x6bhl7gf65zdzhc4-linux-6.6.56
  /nix/store/1y6fh888kww9xqgnm6c46dh3n6jnb816-linux-6.6.56-dev
  /nix/store/q6ps95qsbib1rnz9n9d8h9k65kb5khml-pahole-1.27
  /nix/store/sqipvgf0vwdd7ra956rbvv30wh1cnzs3-python3-minimal-3.12.6
  /nix/store/5wvsiqhdwiypi7kgcl3qbs7mz5icphc3-setup-debug-info-dirs-hook
  /nix/store/zmahjcp2d5pddxi53mhk2fx2mjmhxxy7-source
  /nix/store/d9bdvx6sgigmgsydnr7svrh3w9p8yimv-uname
copying path '/nix/store/zmahjcp2d5pddxi53mhk2fx2mjmhxxy7-source' from 'https://cache.nixos.org'...
copying path '/nix/store/5wvsiqhdwiypi7kgcl3qbs7mz5icphc3-setup-debug-info-dirs-hook' from 'https://cache.nixos.org'...
copying path '/nix/store/9dvhqgc1bqsq7xf8x6bhl7gf65zdzhc4-linux-6.6.56' from 'https://cache.nixos.org'...
copying path '/nix/store/qswh3390mf0shsrbq54md3l8w3jxxcfd-getopt-1.1.6' from 'https://cache.nixos.org'...
copying path '/nix/store/q6ps95qsbib1rnz9n9d8h9k65kb5khml-pahole-1.27' from 'https://cache.nixos.org'...
copying path '/nix/store/wxgwfw64c1rmsi8nhdb0dz8s4haycj6b-elfutils-0.191-bin' from 'https://cache.nixos.org'...
copying path '/nix/store/bbcq7nbvp2wqdjm1j2qxqlmpm3maqnx5-libffi-3.4.6' from 'https://cache.nixos.org'...
copying path '/nix/store/d9bdvx6sgigmgsydnr7svrh3w9p8yimv-uname' from 'https://cache.nixos.org'...
copying path '/nix/store/sqipvgf0vwdd7ra956rbvv30wh1cnzs3-python3-minimal-3.12.6' from 'https://cache.nixos.org'...
copying path '/nix/store/5zc6x2aylvgzi32qna61fshgi6abgvmx-elfutils-0.191-dev' from 'https://cache.nixos.org'...
copying path '/nix/store/1y6fh888kww9xqgnm6c46dh3n6jnb816-linux-6.6.56-dev' from 'https://cache.nixos.org'...
building '/nix/store/iy5pvr57bir5dkgiyhfg529w61f6525s-system76-io-module-1.0.4-6.6.56.drv'...
/nix/store/glibjf1bjrqn0gnmv8ypx9y4x627mbh0-system76-io-module-1.0.4-6.6.56
```

Reply:
Nice, but is that with the sha256 update?  If not then I expect you will see the 1.0.2 code if you look in `/nix/store/zmahjcp2d5pddxi53mhk2fx2mjmhxxy7-source`.

Reply:
No it is with the original, I didn't get an updated sha256 when I ran that command.